### ISSUE TITLE
[Bug Report] Chunks deleted cannot regenerate Piglin Brutes

### ISSUE THREAD

**Issue by Mamox2020** [Amulet-Team/Amulet-Map-Editor#1124]
### Current Behaviour:

When Bastion Remnant chunks are deleted and reloaded, Piglin Brutes do not regenerate.

### Expected behavior:

After testing, when the chunks containing the Ocean Monument are deleted and the structure is reloaded, Elder Guardians are able to regenerate normally. Normally, Elder Guardians and Piglin Brutes both generate alongside their respective structures. Could this be caused by an error in certain areas?

### Environment:

- OS: Windows11
- Minecraft Platform: Bedrock
- Minecraft Version: 1.20.10
- Amulet Version: v0.10.39

**Comment by gentlegiantJGC** [comment]
How the game generates terrain is beyond the scope of Amulet.
There must be some metadata in the chunk that makes the game spawn them but it isn't creating it the second time around.
Have you tried deleting a larger area? The whole structure is generated in one go so you may have to delete the whole structure for it to recreate that data.

**Comment by Mamox2020** [comment]
> How the game generates terrain is beyond the scope of Amulet. There must be some metadata in the chunk that makes the game spawn them but it isn't creating it the second time around. Have you tried deleting a larger area? The whole structure is generated in one go so you may have to delete the whole structure for it to recreate that data.

In the Nether, during chunk operations, I retained only certain chunks and deleted all other chunks, including unloaded chunks. Piglin Brutes in Bastion Remnants within the unloaded chunks were not reset or restored along with the structure.

Using the same method in the End, I retained only the main island chunks and deleted/reset all outer island chunks. When End Cities were reloaded, Shulkers were successfully regenerated alongside the structures.

As previously mentioned, Elder Guardians in Ocean Monuments were also regenerated along with structure resets. Piglin Brutes should follow the same regeneration behavior. I believe this functionality should be achievable with Amulet. Both the End and the Overworld have the capability to restore mobs associated with structures, and the Nether should have this capability as well.
